test/unit/network/ConnectionRegistryTest.js

```diff
@@ -0,0 +1,139 @@
+                    new ConnectionStrategyConfigImpl(),
+                    false,
+                    loadBalancerStub
+                );
+
+                const firstConnection = new sinon.createStubInstance(Connection);
+                connectionRegistry.setConnection(UuidUtil.generate(), firstConnection);
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+
+                const randomConnection = connectionRegistry.getRandomConnection();
+
+                expect(randomConnection).to.be.equal(firstConnection);
+                expect(loadBalancerStub.next.called).to.be.false;
+            }
+        );
+    });
+
+    describe('checkIfInvocationAllowed', function () {
+        it('returns null when ConnectionState=INITIALIZED_ON_CLUSTER and there are some active connections',
+        function () {
+                const connectionRegistry = new ConnectionRegistryImpl(
+                    new ConnectionStrategyConfigImpl(),
+                    false,
+                    new RoundRobinLB()
+                );
+
+                connectionRegistry.setConnectionState(connectionState.INITIALIZED_ON_CLUSTER);
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+
+                expect(connectionRegistry.checkIfInvocationAllowed()).to.be.equal(null);
+            }
+        );
+
+        it('returns ClientOfflineError when ConnectionState=INITIAL and asyncStart=true', function () {
+            const connectionStrategyConfig = new ConnectionStrategyConfigImpl();
+            connectionStrategyConfig.asyncStart = true;
+
+            const connectionRegistry = new ConnectionRegistryImpl(
+                connectionStrategyConfig,
+                false,
+                new RoundRobinLB()
+            );
+
+            connectionRegistry.setConnectionState(connectionState.INITIAL);
+
+            expect(connectionRegistry.checkIfInvocationAllowed()).to.be.instanceof(ClientOfflineError);
+        });
+
+        it('returns IOError when ConnectionState=INITIAL and asyncStart=false', function () {
+            const connectionStrategyConfig = new ConnectionStrategyConfigImpl();
+            connectionStrategyConfig.asyncStart = false;
+
+            const connectionRegistry = new ConnectionRegistryImpl(
+                connectionStrategyConfig,
+                false,
+                new RoundRobinLB()
+            );
+
+            connectionRegistry.setConnectionState(connectionState.INITIAL);
+
+            expect(connectionRegistry.checkIfInvocationAllowed()).to.be.instanceof(IOError);
+        });
+
+        it('returns ClientOfflineError when reconnectMode=ReconnectMode.ASYNC mode, '
+              + 'ConnectionState=INITIALIZED_ON_CLUSTER and there are no connections ',
```

Comment:
```suggestion
              + 'ConnectionState=INITIALIZED_ON_CLUSTER and there are no connections',
```

Reply:
Yes, this is one character shorter, thanks 👍🏻